src/main/common/printf.c

```diff
@@ -38,6 +38,8 @@
 
 #include "build/build_config.h"
 
+#include "common/utils.h"
```

Comment:
You've added `common/utils.h` so potentially you could remove `build/build_config.h`. Can you check if `build_config.h` can be removed for this and all other files where you'v added `utils.h`.